packages/auth-client/src/controllers/engine.ts

```diff
@@ -135,7 +135,7 @@ export class AuthEngine extends IAuthEngine {
       encodeOpts,
     );
 
-    await this.client.requests.set(id, { id, ...cacao });
+    await this.client.requests.set(id, { id, topic: responseTopic, ...cacao });
```

Comment:
This is writing the per-request response topic (which is different to the static pairing topic the dapp/wallet share) instead of the pairingTopic to `requests`.

I think we can simply use `requests.update({ ...cacao })` here since the pending request should already have both the `id` and `pairingTopic` keys.